src/hooks/useAddItem.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+import useFetchItems from './useFetchItems';
+
+const useAddItem = (reference) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState('');
+  const [error, setError] = useState(null);
+  const [duplicateItemMessage, setDuplicateItemMessage] = useState('');
+  const { data: items } = useFetchItems();
+
+  const addItem = async (newItem, timeframe, token, lastPurchased = null) => {
+    setIsLoading(true);
+    try {
+      const ListRef = collection(db, 'Lists');
+      let addedDocument;
+      checkDuplication(newItem)
+        ? showErrorMessage()
+        : (addedDocument = await addDoc(ListRef, {
+            itemName: newItem,
+            timeframe: parseInt(timeframe),
+            lastPurchased,
+            token,
+          }));
+      if (addedDocument) {
+        setIsLoading(false);
+        showSuccesMessage();
+        setError(false);
+      }
```

Comment:
Maybe this can look a bit more simple like this:
```
if (checkDuplication(newItem)) {
showErrorMessage() 
} else {
const addedDocument = ....
... the rest of the things
}
```
The `if(addedDocument)` part only executes if a document is added so I think it makes sense for that code to go in the else block and you can get rid of the `let addedDocument` declaration.
(Not sure if I'm explaining this clearly enough, ask me in slack if you need more info :slightly_smiling_face: )